raiden-ts/src/polyfills.ts

```diff
@@ -28,3 +29,6 @@ request((opts: Record<string, unknown>, cb: ReqCb) => {
 if (!('RTCPeerConnection' in globalThis)) {
   Object.assign(globalThis, require('wrtc')); // eslint-disable-line @typescript-eslint/no-var-requires
 }
+
+// same behavior as python's round()
+BN.set({ ROUNDING_MODE: BN.ROUND_HALF_EVEN });
```

Comment:
This is not `ethers`'s BigNumber, unfortunately. The BigNumber library we use all through our codebase has one nice property for which I want to keep it: it's simple, safe (smaller interface) and integrates seamlessly with `ethers` (as it comes from it).

This one is a **new** dependency (`bignumber.js`), that's why it's called differently (imported as `BN`), to avoid confusion, and we need to include it because, differently from ether's BigNumber, this one supports float/decimal numbers arithmetics.

PC makes a deep usage of float/decimal numbers, several multiplications and even potency operations, followed by `round`s, which make it harder to get the same result with just integer arithmetics (as I wanted). And we need to reproduce these operations precisely, if we want to get scenarios passing, as they test the strict values there.

But I don't want this new library/types in the interfaces, I want to use it only for the calculations where needed and then convert back as soon as possible to our cozy and safer `Int/UInt` types. Fortunately, including it shouldn't increase the bundle size, as it seems it's already pulled in by some other dependency, but if setting this config here globally is a problem for you, I can be explicit about the rounding mode when/where calls to `.dp`/`.toFormat`/`.toFixed` will need it.

Reply:
This got split and fixed in #2581